ch_tools/chadmin/cli/wait_group.py

```diff
@@ -76,29 +85,62 @@ def wait_group():
     default=50.0,
     help="Warning threshold in percent of max_replicated_merges_in_queue.",
 )
-@option(
-    "-v",
-    "--verbose",
-    "verbose",
-    type=int,
-    count=True,
-    default=0,
-    help="Show details about lag.",
-)
 @pass_context
 def wait_replication_sync_command(
-    ctx, status, pause, timeout, xcrit, crit, warn, mwarn, mcrit, verbose
+    ctx,
+    replica_timeout,
+    total_timeout,
+    status,
+    pause,
+    xcrit,
+    crit,
+    warn,
+    mwarn,
+    mcrit,
 ):
-    """Wait for ClickHouse server to sync replication with other replicas using replication-lag command."""
-
-    deadline = time.time() + timeout.total_seconds()
+    """Wait for ClickHouse server to sync replication with other replicas."""
+
+    start_time = time.time()
+    deadline = start_time + total_timeout.total_seconds()
+
+    # Get replicated tables
+    tables = execute_query(
+        ctx,
+        "SELECT name, database FROM system.tables WHERE engine LIKE 'Replicated%' AND engine LIKE '%MergeTree'",
+        format_="JSON",
+    )["data"]
+
+    # Sync tables in cycle
+    for t in tables:
+        full_name = f"{t['database']}.{t['name']}"
+        time_left = deadline - time.time()
+        timeout = min(replica_timeout.total_seconds(), time_left)
+
+        try:
+            execute_query(
+                ctx,
+                f"SYSTEM SYNC REPLICA {full_name}",
+                format_=None,
+                timeout=timeout,
+                settings={"receive_timeout": timeout},
+            )
+        except ReadTimeout:
```

Comment:
Could you cover both of these `except` branches in tests ?

Reply:
Currently we can't control which exception will be raised if equal timeout is set. I thought we can disable something on clickhouse side to wait for ReadTimeout, but it will probably raise other errors like connection timeout, etc.